packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-target-group.ts

```diff
@@ -129,12 +145,31 @@ export class ApplicationTargetGroup extends TargetGroupBase implements IApplicat
   }
 
   /**
-   * Enable sticky routing via a cookie to members of this target group
+   * Enable sticky routing via a cookie to members of this target group.
+   *
+   * Note: If you provide cookieName parameter application-based stickiness attributes (`app_cookie`) will be applied,
+   * otherwise it defaults to duration-based stickiness attributes (`lb_cookie`).
+   *
+   * @see https://docs.aws.amazon.com/elasticloadbalancing/latest/application/sticky-sessions.html
    */
-  public enableCookieStickiness(duration: Duration) {
+  public enableCookieStickiness(duration: Duration, cookieName?: string) {
```

Comment:
Any time we add a second (or third) positional parameter, it's time to step back and consider whether we should alter the signature to take an (extensible) interface instead. For example:

```ts
public enableCookieStickiness(duration: Duration, cookieOpts: CookieOptions) {
```

If in the future, we wanted to add multiple cookie names (for example), an interface would make this signature far more extensible.

Then again, I have a hard time imagining what the extension would be, and it adds non-trivial verbosity overhead depending on the language being used.

What do you think?

Reply:
@njlynch Frankly, I wasn't even sure if adding multiple application cookies was possible.

Edit: 
Now I see that `LoadBalancerAttributes` is an array of attributes (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-loadbalancer-loadbalancerattributes.html). 

I think for the first pass we should only overload the method and see if there is any feedback from the users.

Reply:
I don't think multiple cookies is currently possible; it was just a hypothetical extension.

Reply:
Yeah, I think we should just stick with this implementation for now. Your thoughts?